tests/ra_aid/test_main_limits.py

```
@@ -0,0 +1,232 @@
+        yield mock_repo
+
+
+def test_limit_config_storage(mock_config_repository):
+    """Test that limit arguments are stored in config repository."""
+    import sys
+    from unittest.mock import patch, MagicMock
+    from ra_aid.__main__ import main
+    
+    # Mock repository access functions
+    with patch('ra_aid.__main__.get_config_repository', return_value=mock_config_repository):
+        with patch('ra_aid.__main__.get_trajectory_repository'):
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)